### DISCUSSION TITLE
@jake1164

### DISCUSSION THREAD

**Discussion by Steph1116** [jake1164/Pico-RGB-Matrix-Weather-Clock#44] [General]
I was wondering if you could help me with a code I have using the i75w and matrix board. It's just to understand the code and help me get the time to sit in the centre. I've been to so many discord groups and after spending a few hours, no one really wants to help someone who is new. I'll better explain what I'm looking to do if you wanna help. Thanks

**Comment by jake1164** [comment]
Sure.. email me jake1164@hotmail.com

**Comment by Steph1116** [comment]
Awesome thanks!

Here is the code. I can also email it if you prefer. https://hatebin.com/hbnjysgfkc

**Comment by Steph1116** [comment]
This code came from https://github.com/pimoroni/pimoroni-pico/blob/main/micropython/examples/interstate75/75W/clock.py#L4. I modified it to suit my needs for my second clock. 

So the issues I'm having:

* The time sometimes doesn't correctly display am/pm. When time changes to 12:00am the clock says its 12:00pm and again in reverse. 
*  I can't unplug the unit and plug it back in to a regular USB socket without it resetting the time to 12:00am. I have the network setup, however for some reason it does not run the code.

Your clock however can be plugged into any USB and connect to the network and display the correct stuff.

I'll start with that for now. I know it may be hard for you to run the code without the i75w board. So if there isn't much you can do, its all good. 

Edit; fixed a few things, updated link https://hatebin.com/hbnjysgfkc

Thanks!!

**Reply by jake1164** [reply]
To have the code run without being run though thonny you need to save the file on the pico with the name main.py.  

1. open the file in thonny 
2. save as
3. Raspberry Pi Pico
4. under file name main.py
5. restart device (press restart button)

**Comment by jake1164** [comment]
I actually spent the extra $$ and had one shipped so I have one now.  I'll look at this probably later this week, the whole work thing gets in the way during the beginning of the week.
________________________________

**Comment by Steph1116** [comment]
Not a problem at all. Since I'm running your code on my other device, this project is just for playing around with and learning. I managed to convert the original code from 24 hour to 12-hour so far, but as mentioned I've hit a wall with everything else. And your the only person I sorta know who knows their stuff. But really no hurry one bit.

**Comment by jake1164** [comment]
I would start by creating a repository in github.. this is what github is for.. sharing code.  
Once you make the repository only  you or anybody you authorize can make changes to the code. You also can make it private so only you (and anybody you share it with) can see it.

**Comment by jake1164** [comment]
Do you have 64x32 or 64x64 matrix?

**Comment by Steph1116** [comment]
I have both. But I prefer using the small 64x32 for just a basic digital 12-hour clock. The 64x64 I'm looking to do more fancy stuff so I can have it in the kitchen.

**Comment by Steph1116** [comment]
I emailed you a Discord link to carry on discussions for the interstate75w projects if your interested. I didn't want to clog up this discussion section with that project when it's meant for conversions about your pico w project. I'd like to create a GitHub page for new code for the i75w however I don't have my own code for it.
